File: 2015/go/01.go
package main

import (
	"fmt"
	"io/ioutil"
)

func answer1(input string) int {
	var result int = 0
	for _, char := range input {
		c := string(char)
		if c == "(" {
			result++
		} else if c == ")" {
			result--
		}
	}
	return result
}

func answer2(input string) int {
	var floor int = 0
	for pos, char := range input {
		c := string(char)
		if c == "(" {
			floor++
		} else if c == ")" {
			floor--
		}
		if floor == -1 {
			return pos + 1
		}
	}
	return 0
}

func main() {
	data, _ := ioutil.ReadFile("01input.txt")
	text := string(data)
	fmt.Println(answer1(text))
	fmt.Println(answer2(text))
}

File: 2015/01.lua
local aoc = require "lib.aoc"

local input = aoc.input.getInput()

local function answer1()
	local result = 0
	for _, char in aoc.string.iterate(input) do
		if char == "(" then
			result = result + 1
		elseif char == ")" then
			result = result - 1
		end
	end
	return result
end

local function answer2()
	local floor = 0
	for index, char in aoc.string.iterate(input) do
		if char == "(" then
			floor = floor + 1
		elseif char == ")" then
			floor = floor - 1
		end
		if floor == -1 then
			return index
		end
	end
end

print("answer 1 is " .. answer1())
print("answer 2 is " .. answer2())
